Rocket.Chat/Views/Chat/RCTextView.swift

```diff
@@ -72,28 +72,34 @@ class HighlightLayoutManager: NSLayoutManager {
     func updateCustomEmojiViews() {
         customEmojiViews.forEach { $0.removeFromSuperview() }
         customEmojiViews.removeAll()
+        addCustomEmojiIfNeeded()
+    }
 
+    func addCustomEmojiIfNeeded() {
         message?.enumerateAttributes(in: NSRange(location: 0, length: message.length), options: [], using: { attributes, crange, _ in
             if let attachment = attributes[NSAttributedStringKey.attachment] as? NSTextAttachment {
-
-                guard let position1 = textView.position(from: textView.beginningOfDocument, offset: crange.location) else { return }
-                guard let position2 = textView.position(from: position1, offset: crange.length) else { return }
-                guard let range = textView.textRange(from: position1, to: position2) else { return }
-
-                let rect = textView.firstRect(for: range)
-
-                let emojiView = EmojiView(frame: rect)
-                emojiView.backgroundColor = .white
-                emojiView.isUserInteractionEnabled = false
-
-                if let imageUrlData = attachment.contents,
-                    let imageUrlString = String(data: imageUrlData, encoding: .utf8),
-                    let imageUrl = URL(string: imageUrlString) {
-                    emojiView.emojiImageView.sd_setImage(with: imageUrl, completed: nil)
-                    customEmojiViews.append(emojiView)
+                DispatchQueue.main.async {
+                    guard let position1 = self.textView.position(from: self.textView.beginningOfDocument, offset: crange.location) else { return }
+                    guard let position2 = self.textView.position(from: position1, offset: crange.length) else { return }
+                    guard let range = self.textView.textRange(from: position1, to: position2) else { return }
+
+                    let rect = self.textView.firstRect(for: range)
+
+                    let emojiView = EmojiView(frame: rect)
+                    emojiView.backgroundColor = .white
+                    emojiView.isUserInteractionEnabled = false
+
+                    if let imageUrlData = attachment.contents,
+                        let imageUrlString = String(data: imageUrlData, encoding: .utf8),
+                        let imageUrl = URL(string: imageUrlString) {
+                        DispatchQueue.global(qos: .background).async {
+                            emojiView.emojiImageView.sd_setImage(with: imageUrl, completed: nil)
+                        }
+
+                        self.customEmojiViews.append(emojiView)
+                        self.addSubview(emojiView)
+                    }
                 }
-
-                self.addSubview(emojiView)
```

Comment:
Note that the even when the conditions to add the `EmojiView` to `customEmojiViews` were failing we were still adding the `EmojiView` into the `RCTextView`.